llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -247,6 +247,60 @@ def IREELinalgExt_FftOp : IREELinalgExt_Op<"fft", [
   }];
 }
 
+def IREELinalgExt_ScanOp : IREELinalgExt_Op<"scan", [
+  DeclareOpInterfaceMethods<
+    TiledOpInterface,
+    ["generateScalarImplementation", "getTiledImplementation"]>]> {
+  let summary = "Scan operator";
+  let description = [{
+    Computes the inclusive/exclusive scan along a given dimension.
+  }];
+
+  let arguments = (ins Variadic<AnyType>:$inputs,
+                       Variadic<AnyShaped>:$outputs,
+                       OptionalAttr<I64Attr>:$dimension,
```

Comment:
Instead of making this optional, I'd rather make then required always, and have builder methods that set defaults for these. You can do something like

```
let builders = [
  OpBuilder<(ins "ValueRange":$inputs, "ValueRange":$outputs,
    CArg<"int64_t", "0">:$dimension, CArg<"bool", "true">:$inclusive)>
];
```

Reply:
The defaults are just convenience methods for builders. It will just generate an extra build  method that will set the value of dimension to 0 and inclusive to true if you call that method.

Reply:
I see, makes sene thanks!